src/Views/Stats/StatsJsonView.php

```diff
@@ -225,18 +256,35 @@ private function processSingleSource(array $items): string
 
 		foreach ($items as $item)
 		{
-			// Special case, if the server is empty then change the key to "unknown"
-			if ($this->source === 'server_os' && empty(trim($item[$this->source])))
+			switch ($this->source)
 			{
-				$item[$this->source] = 'unknown';
+				case 'server_os':
+					// Special case, if the server is empty then change the key to "unknown"
+					if (empty(trim($item[$this->source])))
+					{
+						$item[$this->source] = 'unknown';
+					}
+					$data[$this->source][$item[$this->source]] = $item['count'];
+					break;
+
+				case 'cms_php_version':
+					$index = $item['cms_version'] . ' - ' . $item['php_version'];
+					$data[$this->source][$index] = $item['count'];
+					break;
+
+				case 'db_type_version':
+					$index = $item['db_type'] . ' - ' . $item['db_version'];
+					$data[$this->source][$index] = $item['count'];
+					break;
+
+				default:
+					$data[$this->source][$item[$this->source]] = $item['count'];
+					break;
 			}
 
-			$data[$this->source][$item[$this->source]] = $item['count'];
 			$this->totalItems += $item['count'];
 		}
 
-		unset($generator);
```

Comment:
Unsetting the generator seems wrong to me

Reply:
Especially as there is no `$generator` in scope here...

Reply:
? 
not sure to full understand both of you 
i believe i'm doing it correctly,  or not ? 
please elaborate

Reply:
There's no `$generator` var in scope of this method, or for that matter anywhere in this file.  So the syntax is right, but you're unsetting nothing.

Reply:
forgive me, where i'm unsetting  nothing, sorry a bit confused

Reply:
This is being removed @mbabker so it's fine. I missed the variable wasn't defined :)

Reply:
Way to go.